test/regressions/vite.config.mts

```diff
@@ -0,0 +1,90 @@
+import { defineConfig, transformWithEsbuild } from 'vite';
+import react from '@vitejs/plugin-react';
+import { replaceCodePlugin } from 'vite-plugin-replace';
+import * as path from 'path';
+import * as url from 'url';
+
+const currentDirectory = url.fileURLToPath(new URL('.', import.meta.url));
+const WORKSPACE_ROOT = path.resolve(currentDirectory, '../../');
+
+// https://vite.dev/config/
+export default defineConfig({
+  plugins: [
+    {
+      // Unfortunately necessary as we opted to write our jsx in js files
+      name: 'treat-js-files-as-jsx',
+      async transform(code, id) {
+        if (/\/node_modules\//.test(id)) {
+          return null;
+        }
+        if (!/.*\.js$/.test(id)) {
+          return null;
+        }
+        if (id.startsWith('\0')) {
+          return null;
+        }
+        // Use the exposed transform from vite, instead of directly
+        // transforming with esbuild
+        return transformWithEsbuild(code, id, {
+          loader: 'tsx',
+          jsx: 'automatic',
+        });
+      },
+    },
+    react(),
+    replaceCodePlugin({
+      replacements: [
+        {
+          from: '__RELEASE_INFO__',
+          to: 'MTU5NjMxOTIwMDAwMA==',
```

Comment:
Bring back logic in: https://github.com/mui/mui-x/blob/242be42fcdee516942f128bec8d0f5b7d1810cc9/webpackBaseConfig.js#L55